run.py

```diff
@@ -37,9 +38,35 @@ def getDetectorClassConstructors(detectors):
   detectors and returns them in a dict. The dict maps detector name to class
   names. Assumes the detectors have been imported.
   """
-  detectorConstructors = {
-  d : globals()[detectorNameToClass(d)] for d in detectors}
-
+  #handle py2 detectors separately:
+  py2detectors = ",".join([d for d in detectors if (d == "numenta" or d == "numentaTM" or d == "htmjava") ])
+  if py2detectors:
+      ENV_PY2="./pyenv2/bin/python"
+      argss=[]
+      #re-parsing the arguments, as we'll provide it to a new process (python2)
+      if args.numCPUs is not None:
+          argss.append(" --numCPUs="+str(args.numCPUs))
+      argss.append("--skipConfirmation") #always skip confirmation here
+ #     if args.detectors is not None:
+ #       argss.append("--detectors="+py2detectors)
+      if args.dataDir:
+        argss.append("--dataDir="+str(args.dataDir))
+      if args.profilesFile:
+        argss.append("--profilesFile="+str(args.profilesFile))
+      if args.resultsDir:
+        argss.append("--resultsDir="+str(args.resultsDir))
+      if args.windowsFile:
+        argss.append("--windowsFile="+str(args.windowsFile))
+
+  if "htmjava" in py2detectors: # htm.java
+    subprocess.call([ENV_PY2, "./nab/detectors/htmjava/run.py"]+ argss)
```

Comment:
special handling for python2 detectors that allows seamless integration from the main py3 caller.